backend/src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -1487,3 +1487,77 @@ databaseChangeLog:
+            remarks: An API access event that we record for HIPAA auditing.
+            columns:
+              - column:
+                  name: id
+                  type: *idtype
+                  remarks: The internal database identifier for this audit event.
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: event_timestamp
+                  type: DATETIME
+                  remarks: The creation timestamp for this entity.
+                  constraints:
+                    nullable: false
+              - column:
+                  name: request_id
+                  remarks: The correlation ID from the application logs.
+                  type: *string
+              - column:
+                  name: http_request_details
+                  remarks: Key details of the HTTP request.
+                  type: jsonb
+                  constraints:
+                    nullable: false
+              - column:
+                  name: graphql_query_details
+                  remarks: Lightly structured information about the graphql request.
+                  type: jsonb
+              - column:
+                  name: graphql_error_paths
+                  remarks: The list of paths in the graphql response that had errors (usually empty).
+                  type: "text[]"
+              - column:
+                  name: api_user_id
+                  type: *idtype
+                  remarks: The user who performed this action.
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk__api_audit_event__api_user
+                    references: api_user
+              - column:
+                  name: is_admin_user
+                  type: boolean
+                  remarks: Was the user a site administrator at the time the request was made.
+                  constraints:
+                    nullable: false
+              - column:
+                  name: user_permissions
+                  remarks: The permissions attached to the user when the request was made.
+                  type: "text[]"
+              - column:
+                  name: organization_id
+                  type: *idtype
+                  remarks: The organization within which this request was performed (if applicable).
+                  constraints:
+                    nullable: true
+                    foreignKeyName: fk__api_audit_event__organization
+                    references: organization
+              - column:
+                  name: patient_link_id
+                  type: *idtype
+                  remarks: The "magic" patient link that the patient used to perform this action (if applicable).
+                  constraints:
+                    nullable: true
+                    foreignKeyName: fk__api_audit_event__patient_link
+                    references: patient_link
```

Comment:
I still think the standard practice is to put write-heavy logging events into their own data store (e.g. database) for lots of reasons. But I get that it's easier to start by using the same DB especially since it reuses a bunch of code. We should watch for performance issues and maybe monitor the rate of growth of our DB?

Reply:
We should definitely be monitoring that—kicking it to a separate store seems quite possibly needed in the medium term, but wasn't really within scope this week.

Reply:
We can use some Azure Professional Services time to figure out a secondary store if that would be helpful (after this PR, of course). In AWS, I would probably be storing these logs in date-partitioned parquet files in S3, but I don't know if there's a good fire-and-forget ingestion service in Azure (something equivalent to Kinesis Firehose).